pkg/kubectl/cmd/set/set_subjects.go

```diff
@@ -0,0 +1,281 @@
+		return err
+	}
+
+	builder := resource.NewBuilder(o.Mapper, o.Typer, resource.ClientMapperFunc(f.ClientForMapping), f.Decoder(true)).
+		ContinueOnError().
+		NamespaceParam(cmdNamespace).DefaultNamespace().
+		FilenameParam(enforceNamespace, &o.FilenameOptions).
+		Flatten()
+	if !o.Local {
+		builder = builder.
+			SelectorParam(o.Selector).
+			ResourceTypeOrNameArgs(o.All, args...).
+			Latest()
+	}
+
+	o.Infos, err = builder.Do().Infos()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (o *SubjectOptions) Validate() error {
+	if len(o.users) == 0 && len(o.groups) == 0 && len(o.serviceaccounts) == 0 {
+		return fmt.Errorf("you must specify at least one value of user, group or serviceaccount")
+	}
+
+	for _, sa := range o.serviceaccounts {
+		tokens := strings.Split(sa, ":")
+		if len(tokens) != 2 {
+			return fmt.Errorf("serviceaccount must be <namespace>:<name>")
+		}
+	}
+
+	return nil
+}
+
+func (o *SubjectOptions) Run() error {
+	allErrs := []error{}
+	patches := CalculatePatches(o.Infos, o.Encoder, func(info *resource.Info) ([]byte, error) {
+		subjects := []rbac.Subject{}
+		for _, user := range sets.NewString(o.users...).List() {
+			subject := rbac.Subject{
+				Kind:     rbac.UserKind,
+				APIGroup: rbac.GroupName,
+				Name:     user,
+			}
+			subjects = append(subjects, subject)
+		}
+		for _, group := range sets.NewString(o.groups...).List() {
+			subject := rbac.Subject{
+				Kind:     rbac.GroupKind,
+				APIGroup: rbac.GroupName,
+				Name:     group,
+			}
+			subjects = append(subjects, subject)
+		}
+		for _, sa := range sets.NewString(o.serviceaccounts...).List() {
+			tokens := strings.Split(sa, ":")
+			subject := rbac.Subject{
+				Kind:      rbac.ServiceAccountKind,
+				Namespace: tokens[0],
+				Name:      tokens[1],
+			}
+			subjects = append(subjects, subject)
+		}
+
+		transformed, err := updateSubjectForObject(info.Object, subjects, o.Remove)
+		if transformed && err == nil {
+			return runtime.Encode(o.Encoder, info.Object)
+		}
+		return nil, err
+	})
+
+	for _, patch := range patches {
+		info := patch.Info
+		if patch.Err != nil {
+			allErrs = append(allErrs, fmt.Errorf("error: %s/%s %v\n", info.Mapping.Resource, info.Name, patch.Err))
+			continue
+		}
+
+		//no changes
+		if string(patch.Patch) == "{}" || len(patch.Patch) == 0 {
+			allErrs = append(allErrs, fmt.Errorf("info: %s %q was not changed\n", info.Mapping.Resource, info.Name))
+			continue
+		}
+
+		if o.Local || cmdutil.GetDryRunFlag(o.Cmd) {
+			return o.PrintObject(o.Cmd, o.Mapper, info.Object, o.Out)
+		}
+
+		obj, err := resource.NewHelper(info.Client, info.Mapping).Patch(info.Namespace, info.Name, types.StrategicMergePatchType, patch.Patch)
+		if err != nil {
+			allErrs = append(allErrs, fmt.Errorf("failed to patch subjects to rolebinding: %v\n", err))
+			continue
+		}
+		info.Refresh(obj, true)
+
+		if obj, err = resource.NewHelper(info.Client, info.Mapping).Replace(info.Namespace, info.Name, false, obj); err != nil {
```

Comment:
What is this doing? I don't see any recording supported that would make us need to update the object